gremlin/agent_apparmor.profile

```diff
@@ -0,0 +1,107 @@
+#include <tunables/global>
+
+
+profile gremlin-agent flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  network,
+  file,
+  umount,
+  mount,
+
+  /entrypoint.sh rix,
+  /var/lib/gremlin/** rwix,
+  /var/log/gremlin/** rwix,
+  /etc/gremlin/** rwix,
+
+  # /dev/null
+  /dev rix,
+  /dev/null rwlix,
+
+  # Container runtime
+  /run/docker/runtime-runc/moby rwix,
+  /var/run/docker.sock rwix,
+  /run/crio/crio.sock rwix,
+  /run/runc rwix,
+  /run/containers/containers.sock rwix,
+  /run/containerd/runc/k8s.io rwix,
+
+  # We need access to Pid 1's real pid to resolve the container driver
+  # we're just taking a read perm here to accomlish this
+  file r @{PROC}/1/ns/pid,
+  ptrace read,
+
+
+  # Attack capabilities
+  /proc/sysrq-trigger w,
+  /sys/fs/cgroup/** rw,
+  /proc/** rl,
+  # In order to join target container network space
+  @{PROC}/[0-9]+/ns/net w,
+  # In order to assume the root of the target container
+  pivot_root,
+
+  # Needed for specific Gremlin Attacks
+  capability sys_boot,
+  capability sys_time,
+  capability sys_admin, # needed for setns
+  capability net_admin,
+  capability kill,
+  capability setfcap,
+  capability audit_write,
+  capability mknod,
+
+  # Needed to execute attacks
+  capability net_bind_service,
+  capability setuid,
+  capability setgid,
+  capability chown,
```

Comment:
@doshmajhan after much testing I cant get around needing these additional capabilities

Reply:
Huh, I wonder why we need the `net_bind_service`, the other 3 do make sense. Were you able to tell at all during testing why we needed it?

Reply:
`net_bind_service` is a default capability in OCI and I've set the gremlin attacks themselves to inherit these permissions.  I suspect this means runc needs this capability under the covers when it's talking to the container runtime.  The errors I get without it are v unhelpful